src/zenml/stack_stores/rest_stack_store.py

```diff
@@ -0,0 +1,161 @@
+
+    @property
+    def version(self) -> str:
+        """Get the ZenML version."""
+        return Version.parse_obj(self.get("/version")).version
+
+    @property
+    def active_stack_name(self) -> str:
+        """The name of the active stack for this repository."""
+        return ActiveStackName.parse_obj(
+            self.get("/stacks/active")
+        ).active_stack_name
+
+    def activate_stack(self, name: str) -> None:
+        """Activates the stack for the given name."""
+        self.get(f"/stacks/activate/{name}")
+
+    def get_stack_configuration(self, name: str) -> StackConfiguration:
+        """Fetches a stack configuration."""
+        return StackConfiguration.parse_obj(
+            self.get(f"/stacks/configuration/{name}")
+        )
+
+    @property
+    def stack_configurations(self) -> Dict[str, StackConfiguration]:
+        """Configuration for all stacks registered in this repository."""
+        return {
+            key: StackConfiguration.parse_obj(value)
+            for key, value in self.get("/stacks/configurations/")
+        }
+
+    def register_stack_component(
+        self,
+        component: StackComponentConfiguration,
+    ) -> None:
+        """Registers a stack component."""
+        self.post("/components/register", body=component)
+
+    # Custom implementations:
+
+    @property
+    def stacks(self) -> List[StackWrapper]:
+        """All stacks registered in this repository."""
+        return [StackWrapper.parse_obj(s) for s in self.get("/stacks")]
+
+    def get_stack(self, name: str) -> StackWrapper:
+        """Fetches a stack."""
+        return StackWrapper.parse_obj(self.get(f"/stacks/{name}"))
+
+    def register_stack(self, stack: StackWrapper) -> Dict[str, str]:
+        """Registers a stack and its components."""
+        return cast(Dict[str, str], self.post("/stacks/register", stack))
+
+    def deregister_stack(self, name: str) -> None:
+        """Deregisters a stack."""
+        self.get(f"stacks/{name}/deregister")
+
+    def get_stack_component(
+        self, component_type: StackComponentType, name: str
+    ) -> StackComponentConfiguration:
+        """Fetches a registered stack component."""
+        return StackComponentConfiguration.parse_obj(
+            self.get(f"/components/{component_type}/{name}")
+        )
+
+    def get_stack_components(
+        self, component_type: StackComponentType
+    ) -> List[StackComponentConfiguration]:
+        """Fetches all registered stack components of the given type."""
+        return [
+            StackComponentConfiguration.parse_obj(c)
+            for c in self.get(f"/components/{component_type}")
+        ]
+
+    def deregister_stack_component(
+        self, component_type: StackComponentType, name: str
+    ) -> None:
+        """Deregisters a stack component."""
+        self.get(f"/components/deregister/{component_type}/{name}")
+
+    # don't call these! ABC complains that they aren't implemented, but they
```

Comment:
So all the private methods below aren't needed as this interface overwrites the public-facing ones of the base class with it's own logic? Will this be beneficial for us in the long-term or will we frequently run into problems (e.g someone implements a new public method on the `BaseStackStore` which calls one of the unimplemented ones, we'd have to somehow catch that it needs to be implemented in the RestStackStore as well). Or coming from another angle, what's the downside to implementing the private methods instead? More HTTP requests?